### PR TITLE
PWGHF: Adding track selection, changing multiplicity definition from V0M to FT0M

### PR DESCRIPTION
**Pull request by Simple-Shyam**
In this merge request, I changed the definition of multiplicity from V0M to FT0M, and further track selection is applied on the associated particles as suggested by conveners.

### CODE REVIEW

**Review by fcolamar — COMMENTED** [review]
Ciao Shyam, thanks for the updated code, I added a couple of comments.
Also, it's not clear to me why, even if you are only committing the latest changes w.r.t. the previous code version, your commit history starts from about an year ago, including several updates already implemented in the code. Maybe you didn't start from the latest version of the code to prepare the PR?

**Comment by Simple-Shyam** [comment]
> Ciao Shyam, thanks for the updated code, I added a couple of comments. Also, it's not clear to me why, even if you are only committing the latest changes w.r.t. the previous code version, your commit history starts from about an year ago, including several updates already implemented in the code. Maybe you didn't start from the latest version of the code to prepare the PR?

Ciao Fabio, thanks. I started with the latest commit from the official repository. The previous history is not visible because at present the repository is in private mode (most probably). If I make it public then it will be available, and I will do it.

File: PWGHF/HFC/TableProducer/correlatorDplusHadrons.cxx [hunk 1]
```diff
@@ -175,11 +175,12 @@ struct HfCorrelatorDplusHadrons {
   BinningType corrBinning{{binsZVtx, binsMultiplicity}, true};
 
   // Event Mixing for the Data Mode
-  using MySelCollisions = soa::Filtered<soa::Join<aod::Collisions, aod::Mults, aod::DmesonSelection>>;
-  using MyTracks = soa::Filtered<aod::TracksWDca>;
-  using MyCandidatesData = soa::Filtered<soa::Join<aod::HfCand3Prong, aod::HfSelDplusToPiKPi>>;
+  using SelCollisionsWithDplus = soa::Filtered<soa::Join<aod::Collisions, aod::Mults, aod::DmesonSelection>>;
+  using TracksWithData = soa::Filtered<soa::Join<aod::TracksWDca, aod::TrackSelection>>;                       // track Selection applied
+  using TracksWithMc = soa::Filtered<soa::Join<aod::TracksWDca, aod::TrackSelection, o2::aod::McTrackLabels>>; // track Selection applied
```
[the remark this answers is not in the document]
Comment (Simple-Shyam) [reply PWGHF/HFC/TableProducer/correlatorDplusHadrons.cxx]: I don't need it o2::aod::McTrackLabels, I removed it and used only TracksWithDca everywhere (no Data or MC in the name).

File: PWGHF/HFC/TableProducer/correlatorDplusHadrons.cxx [hunk 2]
```diff
@@ -511,44 +518,50 @@ struct HfCorrelatorDplusHadrons {
...
-  void processMcRecMixedEvent(MySelCollisions const& collisions,
-                              MyCandidatesMcRec const& candidates,
-                              MyTracks const& tracks)
+  void processMcRecMixedEvent(SelCollisionsWithDplus const& collisions,
+                              CandidatesDplusMcRec const& candidates,
+                              TracksWithData const& tracks)
   {
     auto tracksTuple = std::make_tuple(candidates, tracks);
-    Pair<MySelCollisions, MyCandidatesMcRec, MyTracks, BinningType> pairMcRec{corrBinning, 5, -1, collisions, tracksTuple, &cache};
+    Pair<SelCollisionsWithDplus, CandidatesDplusMcRec, TracksWithData, BinningType> pairMcRec{corrBinning, 5, -1, collisions, tracksTuple, &cache};
```
[the remark this answers is not in the document]
Comment (Simple-Shyam) [reply PWGHF/HFC/TableProducer/correlatorDplusHadrons.cxx]: Yes applied the same changes here.

**Pull request closed without merging**